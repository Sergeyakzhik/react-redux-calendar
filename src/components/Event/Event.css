.event {
  height: 1.3rem;
  z-index: 1;
  background-color: #ff8201bf;
  border-radius: 7px;
  margin: 0.2rem 0;
  text-align: center;
  left: 0.2rem;
  box-shadow: 4px 4px 5px #00000061;
  user-select: none;
}

.event h4, .event i {
  pointer-events: none;
}

::-webkit-scrollbar {
  width: 5px;
}

::-webkit-scrollbar-track {
  background: white;
  border-radius: 2px;
}

::-webkit-scrollbar-thumb {
  background: #272830;
  border-radius: 2px;
}

::-webkit-scrollbar-thumb:hover {
  background: #555;
}

.delete-button {
  position: absolute;
  left: 0.8rem;
  top: 0.2rem;
  font-size: 1rem;
  color: white;
  transition: color 0.3s;
}

.delete-button:hover {
  color: red;
  transition: color 0.3s;
  cursor: pointer;
}

.table-month .resizer {
  position: absolute;
  right: -17px;
  width: 59px;
  height: 100%;
  top: 0;
  cursor: e-resize;
  background: #0000ff6b;
  z-index: 200;
}

.table-month:active .resizer:hover {
  height: 312px;
  width: 480px;
  top: -431%;
  z-index: 201;
  right: -162px;
}

/* .table-month .dragger {
  position: absolute;
  right: 0;
  width: 82%;
  height: 100%;
  top: 0;
  cursor: grab;
  background: #0000ff6b;
  z-index: 199;
} */

.event:hover {
  cursor: grab;
}

.event:active {
  position: absolute;
  top: 0;
  z-index: 200;
  right: 0;
  cursor: grabbing;
}
