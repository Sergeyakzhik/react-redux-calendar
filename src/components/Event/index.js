import React from 'react';
import './style.css';

let Event = props => (
  <>
    <div className="event" style={props.style}
      targetKey={props.targetKey}
      onMouseEnter={props.onMouseEnter}
      onMouseLeave={props.onMouseLeave}
      onClick={e => e.stopPropagation()}
    >
      <h4>{props.event.name}</h4>
    </div>
    <div className="delete-button" targetKey={props.targetKey} event={props.event} onClick={props.onDeleteButtonClick}>
      <i className="fas fa-times"></i>
    </div>
  </>
);

export default Event;
