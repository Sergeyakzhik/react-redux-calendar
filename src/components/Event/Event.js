import React from 'react';
import './Event.css';

let Event = props => (
  <div>
  </div>
);

export default Event;
