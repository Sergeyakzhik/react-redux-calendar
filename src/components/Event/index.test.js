import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Event from './index';

Enzyme.configure({ adapter: new Adapter() });

describe('Event', () => {
  const props = {
    style: {},
    targetKey: '',
    onMouseEnter: () => {},
    onMouseLeave: () => {},
    onClick: () => {},
    onDeleteButtonClick: () => {},
    event: { name: 'new Event' },
  };
  let component;

  beforeEach(() => {
    component = shallow(<Event {...props} />);
  });

  it('renders Event component', () => {
    expect(component).toMatchSnapshot();
  });

  it('checks event name', () => {
    expect(component.find('h4').text()).toEqual('new Event');
  });
});
