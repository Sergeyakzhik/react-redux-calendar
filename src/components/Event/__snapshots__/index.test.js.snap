// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event render Event component 1`] = `
<Fragment>
  <div
    className="event"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={Object {}}
    targetKey=""
  >
    <h4>
      new Event
    </h4>
  </div>
  <div
    className="delete-button"
    event={
      Object {
        "name": "new Event",
      }
    }
    onClick={[Function]}
    targetKey=""
  >
    <i
      className="fas fa-times"
    />
  </div>
</Fragment>
`;
